workspaces/global-header/plugins/global-header/src/components/GlobalHeader.tsx

```diff
@@ -14,138 +14,19 @@
-    );
-
-  const { menuStates, handleOpen, handleClose } = useDropdownManager();
-
-  const getDropdownButtonProps = (key: string) => ({
-    handleMenu: handleOpen(key),
-    anchorEl: menuStates[key],
-    setAnchorEl: handleClose(key),
-  });
-
-  const getIconButtonProps = (props: Record<string, any>) => ({
-    icon: props.icon ?? '',
-    tooltip: props.tooltip ?? '',
-    to: props.to ?? '',
-  });
-  const renderComponents = (mountPoints: GlobalHeaderComponentMountPoint[]) =>
-    mountPoints.map((mp, index) => {
-      let displayHeaderIcon = false;
-      if (frontendData) {
-        for (const pluginData of Object.values(frontendData)) {
-          const dynamicRoutes = pluginData.dynamicRoutes ?? [];
-          if (
-            dynamicRoutes.some(
-              (route: { path: string }) => route.path === mp.config?.props?.to,
-            )
-          ) {
-            displayHeaderIcon = true;
-            break;
-          }
-        }
-      }
-
-      switch (mp.config?.type) {
-        case ComponentType.SEARCH:
-          return (
-            <ErrorBoundary>
-              {/* eslint-disable-next-line react/no-array-index-key */}
-              <mp.Component key={index} />
-            </ErrorBoundary>
-          );
-        case ComponentType.DROPDOWN_BUTTON:
-          return (
-            <ErrorBoundary>
-              <mp.Component
-                // eslint-disable-next-line react/no-array-index-key
-                key={index}
-                {...getDropdownButtonProps(mp.config?.key ?? index.toString())}
-                {...mp.config?.props}
-              />
-            </ErrorBoundary>
-          );
-        case ComponentType.ICON_BUTTON:
-          return (
-            <ErrorBoundary>
-              {displayHeaderIcon && (
-                <mp.Component
-                  // eslint-disable-next-line react/no-array-index-key
-                  key={index}
-                  {...getIconButtonProps(mp.config?.props ?? {})}
-                />
-              )}
-            </ErrorBoundary>
-          );
-        default:
-          return null;
-      }
-    });
-
   return (
-    <AppBar
-      position="sticky"
-      component="nav"
-      sx={{
-        backgroundColor: theme.palette.mode === 'dark' ? '#1b1d21' : '#212427',
-        backgroundImage: 'none',
-      }}
-    >
-      <Toolbar>
-        {renderComponents(globalHeaderStartComponentsMountPoints)}
-        {globalHeaderStartComponentsMountPoints.length > 0 &&
-          globalHeaderEndComponentsMountPoints.length > 0 && (
-            <Divider
-              orientation="vertical"
-              flexItem
-              sx={{ borderColor: '#4F5255', marginX: 1 }}
-            />
-          )}
-        {renderComponents(globalHeaderEndComponentsMountPoints)}
-      </Toolbar>
-    </AppBar>
+    <GlobalHeaderComponent
+      globalHeaderMountPoints={allGlobalHeaderMountPoints ?? []}
+      supportUrl={supportUrl ?? ''}
```

Comment:
Nit: Why not make the prop optional and pass undefined when its not configured. I personally think that defaulting to empty strings brings up a (small) risk.